src/Core/Import/File/FileUploader.php

```diff
@@ -0,0 +1,130 @@
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Import\File;
+
+use PrestaShop\PrestaShop\Core\Import\ImportDirectory;
+use PrestaShopBundle\Exception\FileUploadException;
+use Symfony\Component\HttpFoundation\File\File;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * FileUploader is responsible for uploading import files to import directory
+ */
+final class FileUploader
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var ImportDirectory
+     */
+    private $importDirectory;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param ImportDirectory $importDirectory
+     */
+    public function __construct(
+        TranslatorInterface $translator,
+        ImportDirectory $importDirectory
+    ) {
+        $this->translator = $translator;
+        $this->importDirectory = $importDirectory;
+    }
+
+    /**
+     * Handle import file uploading to admin import/ directory
+     *
+     * @param UploadedFile $uploadedFile
+     *
+     * @return File
+     *
+     * @throws FileUploadException
+     */
+    public function upload(UploadedFile $uploadedFile)
+    {
+        if ($error = $this->validateUploadedFile($uploadedFile)) {
+            throw new FileUploadException($error);
+        }
+
+        $uploadedFileName = sprintf(
+            '%s-%s',
+            date('YmdHis'),
+            $uploadedFile->getClientOriginalName()
+        );
+
+        $file = $uploadedFile->move(
+            $this->importDirectory,
+            $uploadedFileName
+        );
+
+        return $file;
+    }
+
+    /**
+     * Check if uploaded file is valid
+     *
+     * @param UploadedFile $uploadedFile
+     *
+     * @return string|false     Returns error string on error or FALSE otherwise
+     */
+    protected function validateUploadedFile(UploadedFile $uploadedFile)
+    {
+        $error = false;
+
+        switch ($uploadedFile->getError()) {
+            case UPLOAD_ERR_INI_SIZE:
+                $error = $this->translator->trans('The uploaded file exceeds the upload_max_filesize directive in php.ini. If your server configuration allows it, you may add a directive in your .htaccess.', [], 'Admin.Advparameters.Notification');
+                break;
+            case UPLOAD_ERR_FORM_SIZE:
+                $error = $this->translator->trans('The uploaded file exceeds the post_max_size directive in php.ini. If your server configuration allows it, you may add a directive in your .htaccess, for example:', [], 'Admin.Advparameters.Notification')
+                    .'<br/><a href="#" >
+					<code>php_value post_max_size 20M</code> '.
```

Comment:
@mickaelandrieu do you have any suggestion of how to deal with error messages that includes html tags?

Reply:
I don't think we are doing that already, how about reword the translation key?

`The uploaded file exceeds the post_max_size directive in php.ini. If your server configuration allows it, you may add a directive in your .htaccess, "php_value post_max_size" for example.'`

Reply:
@mickaelandrieu, it seems clearer. I'm okay if you want to add it.